libcloud/common/base.py

```diff
@@ -46,11 +48,48 @@
 
 from libcloud.utils.misc import lowercase_keys
 from libcloud.utils.compression import decompress_data
+from libcloud.common.exception import RateLimit, exception_from_message
 from libcloud.common.types import LibcloudError, MalformedResponseError
-
 from libcloud.httplib_ssl import LibcloudHTTPConnection
 from libcloud.httplib_ssl import LibcloudHTTPSConnection
 
+DEFAULT_TIMEOUT = 3
```

Comment:
So I just realized that there's no way to override these values because of how you're doing things with a decorator.  The person using libcloud needs to be be able to configure these values when they create a driver.  You might have to rethink using a decorator or figure out a way to dynamically inject the values into it.